packages/swirl-components/src/components/swirl-image-grid-item/swirl-image-grid-item.tsx

```diff
@@ -102,19 +103,20 @@ export class SwirlImageGridItem {
       return;
     }
 
-    this.intersectionObserver = new IntersectionObserver(
-      this.onVisibilityChange.bind(this),
-      {
-        threshold: 0,
-      }
-    );
+    this.intersectionObserver = new IntersectionObserver((entries) => {
+      this.handleIntersectionEntries(entries);
+    });
 
     this.intersectionObserver.observe(this.el);
   }
 
-  private onVisibilityChange(entries: IntersectionObserverEntry[]) {
-    this.inViewport = entries.some((entry) => entry.isIntersecting);
-  }
+  private handleIntersectionEntries = debounce(
+    (entries: IntersectionObserverEntry[]) => {
+      const sorted = [...entries].sort((a, b) => a.time - b.time);
+      this.inViewport = sorted[sorted.length - 1].isIntersecting;
```

Comment:
Just personal preference. Ignore if you disagree. `sorted.at(-1).isIntersecting`

Reply:
agreed